jsonnet/components/thanos-ruler.libsonnet

```diff
@@ -1,495 +1,485 @@
+                name: 'secret-thanos-ruler-tls',
+              },
+              {
+                mountPath: '/etc/tls/grpc',
+                name: 'secret-grpc-tls',
+              },
+              {
+                mountPath: '/etc/prometheus/configmaps/serving-certs-ca-bundle',
+                name: 'serving-certs-ca-bundle',
+              },
+            ],
+          },
+          {
+            name: 'thanos-ruler-proxy',
+            image: 'quay.io/openshift/oauth-proxy:latest',  //FIXME(paulfantom)
+            ports: [{
+              containerPort: $.service.spec.ports[0].port,
+              name: 'web',
+            }],
+            env: [
+              { name: 'HTTP_PROXY', value: '' },
+              { name: 'HTTPS_PROXY', value: '' },
+              { name: 'NO_PROXY', value: '' },
+            ],
+            args: [
+              '-provider=openshift',
+              '-https-address=:9091',
+              '-http-address=',
+              '-email-domain=*',
+              '-upstream=http://localhost:10902',
+              '-openshift-sar={"resource": "namespaces", "verb": "get"}',
+              '-openshift-delegate-urls={"/": {"resource": "namespaces", "verb": "get"}}',
+              '-tls-cert=/etc/tls/private/tls.crt',
+              '-tls-key=/etc/tls/private/tls.key',
+              '-client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token',
+              '-cookie-secret-file=/etc/proxy/secrets/session_secret',
+              '-openshift-service-account=thanos-ruler',
+              '-openshift-ca=/etc/pki/tls/cert.pem',
+              '-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
+            ],
+            terminationMessagePolicy: 'FallbackToLogsOnError',
+            resources: {
+              requests: {
+                cpu: '1m',
+                memory: '12Mi',
+              },
             },
+            volumeMounts: [
+              {
+                mountPath: '/etc/tls/private',
+                name: 'secret-thanos-ruler-tls',
+              },
+              {
+                mountPath: '/etc/proxy/secrets',
+                name: 'secret-thanos-ruler-oauth-cookie',
+              },
+            ],
           },
-        },
-      ],
+          {
+            // Note: this is performing strategic-merge-patch for config-reloader container.
+            // Remainder of the container configuration is managed by prometheus-operator based on $.thanosRuler.spec
+            name: 'config-reloader',
+            resources: {
+              requests: {
+                cpu: '1m',
+                memory: '10Mi',
+              },
+            },
+          },
+        ],
+      },
     },
-  },
 
-  podDisruptionBudget: {
-    apiVersion: 'policy/v1',
-    kind: 'PodDisruptionBudget',
-    metadata: {
-      name: 'thanos-ruler-' + cfg.name,
-      namespace: cfg.namespace,
-      labels: {
-        thanosRulerName: cfg.name,
+    podDisruptionBudget: {
```

Comment:
there's no way to manage PDB in kube-thanos? If not, it would seem like a good addition

Reply:
not in ruler libsonnet. We could propose it and I'm happy to contribute that, but I'm not sure it should block this PR? Or do we prefer to tackle it as one?

Reply:
agreed that it doesn't block this PR as long as we have a proper TODO comment.